plugin/helpers_bots.go

```diff
@@ -4,12 +4,70 @@
 package plugin
 
 import (
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/pkg/errors"
 )
 
-func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error) {
+type EnsureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*EnsureBotOptions)
+
+func ProfileImagePath(path string) EnsureBotOption {
+	return func(args *EnsureBotOptions) {
+		args.ProfileImagePath = path
+	}
+}
+
+func IconImagePath(path string) EnsureBotOption {
+	return func(args *EnsureBotOptions) {
+		args.IconImagePath = path
+	}
+}
+
+func (p *HelpersImpl) readFile(path string) ([]byte, error) {
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get bundle path")
+	}
+
+	imageBytes, err := ioutil.ReadFile(filepath.Join(bundlePath, path))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read image")
+	}
+	return imageBytes, nil
+}
+
+func (p *HelpersImpl) EnsureBot(bot *model.Bot, options ...EnsureBotOption) (retBotId string, retErr error) {
+	// Default options
+	o := &EnsureBotOptions{
+		ProfileImagePath: "",
```

Comment:
i think it's fine not to double-reset them, as they're already initialized to `""`. Just do `o := &EnsureBotOptions{}` instead

Reply:
Apologies for missing this the first time. :( minor nit though.

Reply:
Actually I like the previous code more because it made clear, what the default values are. 2/5 @ali-farooq0 thoughts?

Reply:
@ali-farooq0 Any update on this?

Reply:
@hanzei if it had non-zero default values I would've agreed. It's quite common in go not to set zero values. 
But 0/5 on this though, I'll let you make the final call, personally good with either 👌 

https://dave.cheney.net/2013/01/19/what-is-the-zero-value-and-why-is-it-useful

Reply:
@alxsah Would you please add the longer version back? I would like to be explicit here.

Reply:
@hanzei Done 😄